### ISSUE TITLE
kvm instance with spice configured starts with kvm option "-nographic"

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#278]
Originally reported of Google Code with ID 222.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

gnt-cluster (ganeti v2.5.0rc5) 2.5.0~rc5

Software version: 2.5.0~rc5
Internode protocol: 2050000
Configuration format: 2050000
OS api version: 20
Export interface: 0


<b>What distribution are you using?</b>
Debian Squeeze 
# kvm --version
QEMU emulator version 1.0 (qemu-kvm-1.0 Debian 1.0+dfsg-8~bpo60+1), Copyright (c) 2003-2008 Fabrice Bellard


<b>What steps will reproduce the problem?</b>
1. configure a kvm instance with "spice_bind=0.0.0.0"
2. start the instance
3. ps output of kvm process shows the "-nographic" option is used

<b>What is the expected output? What do you see instead?</b>

<b>Please provide any additional information below.</b>

hv_kvm.py seems to handle the spice case incorrectly.

line 676 tests for "vnc_bind_addess" and else (line 711) it adds "-nographic" on the kvm command line, but it misses to handle the case where spice is used.

```
Originally added on 2012-03-21 13:44:47 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This patch fixes the issue:


--- lib/hypervisor/hv_kvm.py	2012-01-10 15:22:49.000000000 +0100
+++ lib/hypervisor/hv_kvm.py.new	2012-03-22 14:50:27.000000000 +0100
@@ -707,8 +707,6 @@
         vnc_arg = "unix:%s/%s.vnc" % (vnc_bind_address, instance.name)
 
       kvm_cmd.extend(["-vnc", vnc_arg])
-    else:
-      kvm_cmd.extend(["-nographic"])
 
     monitor_dev = ("unix:%s,server,nowait" %
                    self._InstanceMonitor(instance.name))
@@ -768,6 +766,10 @@
       logging.info("KVM: SPICE will listen on port %s", instance.network_port)
       kvm_cmd.extend(["-spice", spice_arg])
 
+    # If neither spice or vnc graphics is used, add "-nographic"
+    if not spice_bind or vnc_bind_address:
+      kvm_cmd.extend(["-nographic"])
+
     if hvp[constants.HV_USE_LOCALTIME]:
       kvm_cmd.extend(["-localtime"])

```
Originally added on 2012-03-22 13:50:56 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

Thanks for the patch. This was fixed in a patch in stable-2.5 earlier this week, and will be released as part of rc6. Thanks a lot for reporting it and looking into it.

Guido
```
Originally added on 2012-03-23 12:38:33 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Migrate old issues from 'fixed' to 'released'.
```
Originally added on 2012-10-22 09:16:17 +0000 UTC.

Changed State: Released
